pay-api/src/pay_api/services/base_payment_system.py

```diff
@@ -306,7 +306,7 @@ def _refund_and_create_credit_memo(
             remaining_amount=refund_amount,
             account_id=invoice.payment_account_id,
             created_invoice_id=invoice.id,
-        ).flush()
+        ).save()
```

Comment:
This should be save, because the credit is already created in CMS

I was running into weird issues with `flush` to do with deadlocks  it would trigger a large query it seemed like, this should be save anyways